src/components/microworld/microworld.jsx

```diff
@@ -0,0 +1,251 @@
+var classNames = require('classnames');
+var React = require('react');
+var render = require('../../lib/render.jsx');
+var omit = require('lodash.omit');
+
+require('./microworld.scss');
+
+var Box = require('../../components/box/box.jsx');
+var Carousel = require('../../components/carousel/carousel.jsx');
+var Modal = require('../../components/modal/modal.jsx');
+var TipsSlider = require('../../components/tipsslider/tipsslider.jsx');
+
+var Microworld = React.createClass({
+    type: 'Microworld',
+    propTypes: {
+        microworldData: React.PropTypes.node.isRequired
+    },
+    showVideo: function (key) {
+        var videoOpenArr = this.state.videoOpen;
+        videoOpenArr[key] = true;
+        this.setState({videoOpen: videoOpenArr});
+    },
+    closeVideo: function (key) {
+        var videoOpenArr = this.state.videoOpen;
+        videoOpenArr[key] = false;
+        this.setState({videoOpen: videoOpenArr});
+    },
+    getInitialState: function () {
+        return {
+            videoOpen: {},
+            featuredGlobal: {},
+            featuredLocal: {},
+            microworld_data: this.props.microworldData
```

Comment:
I think this could be moved to props instead of state if it's not going to be changed within the component itself